src/Extrapolators/BlockedExtrapolator.cpp

```diff
@@ -413,33 +423,53 @@ void BlockedExtrapolator<SQLQuerierType, GraphType, AnnouncementType, ASType>::s
             if (ann.second.priority < 200) {
                 continue;
             }
-            
-            // Set the priority of the announcement at destination 
-            uint32_t old_priority = ann.second.priority;
-            uint32_t path_len_weight = old_priority % 100;
-            if (path_len_weight == 0) {
-                // For MRT ann at origin: old_priority = 400
-                path_len_weight = 99;
-            } else {
-                // Sub 1 for the current hop
-                path_len_weight -= 1;
-            }
-            uint32_t priority = 200 + path_len_weight;
-            
-            AnnouncementType temp = AnnouncementType(ann.second);
-            temp.priority = priority;
-            temp.from_monitor = false;
-            temp.received_from_asn = asn;
-
-            // Push announcement with new priority to ann vector
-            // anns_to_providers.push_back(AnnouncementType(ann.second.origin,
-            //                                          ann.second.prefix.addr,
-            //                                          ann.second.prefix.netmask,
-            //                                          priority,
-            //                                          asn,
-            //                                          ann.second.tstamp));
 
-            anns_to_providers.push_back(temp);
+            // Automatic multihomed mode
+            // Propagate an announcement to providers if there are some that haven't received an announcement from that prefix with origin = current asn
+            uint32_t providers_with_ann = 0;
+            if (mh_mode == 1) {
+                // Check if AS is multihomed
+                if (source_as->customers->empty()) {
+                    // Check if all providers have the announcement
+                    for (uint32_t provider_asn : *source_as->providers) {
+                        auto *recving_as = this->graph->ases->find(provider_asn)->second;
+                        for (auto &recving_as_ann : *recving_as->all_anns) {
+                            if (recving_as_ann.second.origin == ann.second.origin && recving_as_ann.second.prefix.addr == ann.second.prefix.addr && recving_as_ann.second.prefix.netmask == ann.second.prefix.netmask) {
+                                providers_with_ann++;
+                            }
+                        }
+                    }
+                }
+            } 
+            // If there are providers that don't have the announcement, propagate to providers
+            if (providers_with_ann < source_as->providers->size()) {
```

Comment:
The propagation to providers should happen if _no_ providers received it, if some received it then we should not propagate (assuming this indicates a preference towards a provider like in a primary/backup configuration). 
```suggestion
            if (providers_with_ann == 0) {
```

Reply:
Done